gateway/src/apicast/policy/caching/apicast-policy.json

```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "http://apicast.io/policy-v1/schema#manifest#",
+  "name": "Caching policy",
+  "description":
+    ["Configures a cache for the authentication calls against the 3scale ",
+     "backend. This policy support three kinds of caching: \n",
+     " - Strict: it only caches authorized calls. Denied and failed calls ",
+     "invalidate the cache entry.\n",
+     " - Resilient: caches authorized and denied calls. Failed calls do not ",
+     "invalidate the cache. This allows us to authorize and deny calls ",
+     "according to the result of the last request made even when backend is ",
+     "down.\n",
+     "- Allow: caches authorized and denied calls. When backend is ",
+     "unavailable, it will cache an authorization. In practice, this means ",
+     "that when backend is down _any_ request will be authorized unless last ",
+     "call to backend for that request returned 'deny' (status code = 4xx). ",
+     "Make sure to understand the implications of that before using this ",
+     "mode. It makes sense only in very specific use cases.\n",
+     "- None: disables caching."],
+  "version": "0.1",
+  "configuration": {
```

Comment:
We might need to force `$schema` key in this object, so it can be easier for other libraries to use this and know what schema it is.

Reply:
No need to force the `$schema` key, the lib figure this out.

Reply:
I realized that the `lint-schema` target is not working correctly. It's not checking that `configuration`complies with `http://json-schema.org/draft-07/schema#"`.
Adding `$schema` in the configuration object does not solve the issue.

Reply:
I was using https://www.jsonschemavalidator.net for validation and it was ok.
But now it fails on missing `$id` in the `definitions/id` and `definitions/schema`. 
Just adding there `"$id": "#/definitions/schema",` and `"$id":  "#/definitions/version",` makes it validate in that online editor.

What other field was not validated correctly? I tried to change the "type" column from "string" to some invalid value and that is caught.

Reply:
@mikz When I run `make lint-schema`, the error you mention is not detected. Not sure why.
I added those ids in your commit.

Another thing I noticed is that, if I remove the "type" attr from `caching_type` in the caching policy config, https://mozilla-services.github.io/react-jsonschema-form/ complains, but `make lint-schema` does not.